subsys/dfu/boot/mcuboot.c

```diff
@@ -164,115 +162,123 @@ static const struct boot_swap_table boot_swap_tables[] = {
 };
 #define BOOT_SWAP_TABLES_COUNT (ARRAY_SIZE(boot_swap_tables))
 
-static struct device *flash_dev;
-
-static int boot_flag_offs(int flag, u32_t bank_offs, u32_t *offs)
+static int boot_flag_offs(int flag, const struct flash_area *fa, u32_t *offs)
 {
 	switch (flag) {
 	case BOOT_FLAG_COPY_DONE:
-		*offs = COPY_DONE_OFFS(bank_offs);
+		*offs = COPY_DONE_OFFS(fa);
 		return 0;
 	case BOOT_FLAG_IMAGE_OK:
-		*offs = IMAGE_OK_OFFS(bank_offs);
+		*offs = IMAGE_OK_OFFS(fa);
 		return 0;
 	default:
 		return -ENOTSUP;
 	}
 }
 
-static int boot_flash_write(off_t offs, const void *data, size_t len)
+static int boot_flag_write(int flag, u8_t bank_id)
 {
+	const struct flash_area *fa;
+	u8_t buf[FLASH_MIN_WRITE_SIZE];
+	u32_t offs;
 	int rc;
 
-	rc = flash_write_protection_set(flash_dev, false);
+	rc = flash_area_open(bank_id, &fa);
 	if (rc) {
 		return rc;
 	}
 
-	rc = flash_write(flash_dev, offs, data, len);
-	if (rc) {
-		return rc;
-	}
-
-	rc = flash_write_protection_set(flash_dev, true);
-
-	return rc;
-}
-
-static int boot_flag_write(int flag, u32_t bank_offs)
-{
-	u8_t buf[FLASH_MIN_WRITE_SIZE];
-	u32_t offs;
-	int rc;
-
-	rc = boot_flag_offs(flag, bank_offs, &offs);
+	rc = boot_flag_offs(flag, fa, &offs);
 	if (rc != 0) {
+		flash_area_close(fa);
 		return rc;
 	}
 
 	memset(buf, BOOT_FLAG_UNSET, sizeof(buf));
 	buf[0] = BOOT_FLAG_SET;
 
-	rc = boot_flash_write(offs, buf, sizeof(buf));
+	rc = flash_area_write(fa, offs, buf, sizeof(buf));
+	flash_area_close(fa);
```

Comment:
honestly I don't care about closing flash areas in zephyr-code as it is concept form mynew, and now no one know whom it serve.

Reply:
Are you saying that open and close do not need to be in pairs?

Reply:
yes, exactly that - you can decide to keep your code as is. (flash_are_cloce is nop )